test/e2e/traceflow_test.go

```diff
@@ -320,3 +331,19 @@ func (data *TestData) createNPAllowAllEgress(name string) (*networkingv1.Network
 	}
 	return data.createNetworkPolicy(name, spec)
 }
+
+// waitForNetworkpolicyRealized waits for the NetworkPolicy to be realized by the antrea-agent Pod.
+func (data *TestData) waitForNetworkpolicyRealized(pod string, networkpolicy string) error {
+	if err := wait.Poll(200*time.Millisecond, 5*time.Second, func() (bool, error) {
+		cmds := []string{"antctl", "get", "networkpolicy", networkpolicy, "-n", testNamespace}
+		if _, _, err := runAntctl(pod, cmds, data); err != nil {
+			return false, nil
+		}
+		return true, nil
+	}); err == wait.ErrWaitTimeout {
+		return fmt.Errorf("NetworkPolicy %s isn't realized in time", networkpolicy)
+	} else if err != nil {
+		return fmt.Errorf("Error when executing antctl get NetworkPolicy: %v", err)
```

Comment:
At the moment, we never return an error from inside `Poll`, so this if case is never going to run.

So it seems there is no way to distinguish between "antctl could not run" and "antctl could not find the policy"? That's a bit of an issue here IMO, but ultimately that comes down to `runCommandFromPod` which does not make that distinction. I don't know if there is an easy way to update `runCommandFromPod` so that we can tell these 2 cases apart?

The goal here would be to abort early if `antctl` cannot be run because of some issue with the test setup.

Reply:
Yes. Since `runCommandFromPod` uses client-go functions so I think the easy way is to detect if stderr contains `the server could not find the requested resource`.
@tnqn How do you think?

Reply:
I'm ok with current approach, but should it return `false, nil` if it's NotFound error? otherwise it wouldn't retry

Reply:
Yes. If error is not NotFound, then it should `return false, err`. Updated.